src/meltano/core/job_state.py

```diff
@@ -160,7 +161,7 @@ def merge_partial(
         Args:
             state: the partial state to merge onto this state.
         """
-        self.partial_state = merge(
```

Comment:
Removed this redundant assignment cause `merge` mutates the second argument in-place. This mutability is seriously bug-prone but changing the behavior is a larger task.